app/src/main/java/com/duckduckgo/app/notification/NotificationScheduler.kt

```diff
@@ -103,11 +150,114 @@ class NotificationScheduler @Inject constructor(
             intent.type = eventType
             intent.putExtra(PIXEL_SUFFIX_EXTRA, specification.pixelSuffix)
             intent.putExtra(NOTIFICATION_SYSTEM_ID_EXTRA, specification.systemId)
+            intent.putExtra(NOTIFICATION_AUTO_CANCEL, specification.autoCancel)
+            return getService(context, 0, intent, 0)!!
+        }
+    }
+
+    class SearchPromptNotificationWorker(val context: Context, val params: WorkerParameters) : CoroutineWorker(context, params) {
+        lateinit var manager: NotificationManagerCompat
+        lateinit var factory: NotificationFactory
+        lateinit var notificationDao: NotificationDao
+        lateinit var notification: SearchNotification
+        lateinit var pixel: Pixel
+
+        override suspend fun doWork(): Result {
```

Comment:
There's some logic I would test here, I saw that previous `worker` are not tested, but I think we should cover any domain logic. (e.g: if we show notification, we insert notification id in the database.)

Reply:
ditto